Req 1 Create

lucasferrarisoares opened the pull request:
Classe médico e suas depedências criadas.

Função "admitPatient" editada: agora ao internar um paciente, o usuário precisa também informar o id do doutor reponsável pela internação.

Criada a função "changeDoctor": é possível trocar o doutor responsável por uma internação, desde que ela ainda esteja ativa, ou seja, que o paciente não tenha recebido alta.

lucasferrarisoares committed:
Criando classe médico

lucasferrarisoares committed:
Update DoctorModel.java

lucasferrarisoares committed:
Finalizado requisição

Parte base da requisição finalizada, agora entro na fase de revisão/melhoria do código

lucasferrarisoares committed:
Update AdmissionLogService.java

lucasferrarisoares committed:
Criado Histórico de médicos por internação.

mathpablo commented on src/main/java/br/com/ControleDePacientes/model/DoctorModel.java:
```diff
@@ -0,0 +1,25 @@
+    @Enumerated(EnumType.STRING)
+    @Column(nullable = false)
+    private SpecialtyEnum specialty;
+}
```
Acho que poderia ter um atributo para salvar o nome do médico.

mathpablo commented on src/main/java/br/com/ControleDePacientes/repository/DoctorRepository.java:
```diff
@@ -0,0 +1,13 @@
+
+    @Query("SELECT d FROM DoctorModel d " +
+            "WHERE d.crm = :crm")
+    DoctorModel findDoctorByCRM(@Param("crm") Long crm);
```
aqui talvez poderira colocar  um  Optional<DoctorModel>  para evitar problemas com o null.

mathpablo commented on src/main/java/br/com/ControleDePacientes/controller/DoctorController.java:
```diff
@@ -0,0 +1,41 @@
+    }
+
+    @DeleteMapping("/{id}")
+    public void deleteDoctor(Long id){
```
Acho que poderia colocar um @PathVariable nas rotas com parametros

mathpablo commented on src/main/java/br/com/ControleDePacientes/service/DoctorService.java:
```diff
@@ -0,0 +1,48 @@
+        doctor.setSpecialty(updateddoctor.getSpecialty());
+
+        return this.doctorRepository.save(doctor);
+    }
```
Poderia fazer para atualizar o crm tambem

lucasferrarisoares committed:
adicionado nome a classe Doctor

lucasferrarisoares committed:
Mudanças pós reunião

Mudança no nome do atributo que contem um DoctorModel dentro de AdmissionLogModel
Nome do médico adicionado em DoctorModel
Colocado ResponseEntity no return do controller de DoctorController
Optional no retorno de DoctorRepository
Mudança de DoctorHistory... para DoctorLog para uma padronização do projeto

CarlosFavarao commented on src/main/java/br/com/ControleDePacientes/service/AdmissionLogService.java:
```diff
@@ -89,4 +100,32 @@ public Page<LogDTO> getAdmissionHistoryByPatientId(Long patientId, Pageable page
     public Page<BedHistoryDTO> getBedAdmissionHistory(Long bedId, Pageable pageable){
         return this.admissionLogRepository.findBedAdmissionHistoryById(bedId, pageable);
     }
+
```
Adicionar um @Transactional aqui.
